src/server/frontend_xwayland/xwayland_wm.cpp

```diff
@@ -298,7 +298,14 @@ void mf::XWaylandWM::handle_events()
 
     while (xcb_generic_event_t* const event = xcb_poll_for_event(xcb_connection))
     {
-        handle_event(event);
+        try
+        {
+            handle_event(event);
+        }
+        catch (std::exception const& e)
+        {
+            log_error(e.what());
```

Comment:
It seems wrong to log an "error" when we're going to carry on by ignoring it.

How about:
```c++
        catch (...)
        {
            log(logging::Severity::warning,
                MIR_LOG_COMPONENT,
                std::current_exception(),
                "Failed to handle xcb event.");
        }
```

Reply:
Sure, done